numpy/core/arrayprint.py

```diff
@@ -507,14 +530,9 @@ def array2string(a, max_line_width=None, precision=None,
         formatter = _formatter
 
     if a.shape == ():
-        x = a.item()
-        if a.dtype.fields is not None:
-            arr = array([x], dtype=a.dtype)
-            format_function = _get_format_function(
-                    arr, precision, suppress_small, formatter)
-            lst = format_function(arr[0])
-        else:
-            lst = style(x)
+        format_function = _get_format_function(
+                a.reshape((1,)), precision, suppress_small, formatter)
+        lst = format_function(a[()])
```

Comment:
You previously suggested this case could be moved to `_array2string` in an `if rank == 0` case, but I like it here to emphasize it is treated specially.